### PR TITLE
Use resize observer to simplify use get plot

### PR DESCRIPTION
**Pull request by mattseddon**
Matches part of the change that I made in https://github.com/iterative/studio/pull/8490.

@sroy3 when you're back I'd like to get your opinion on performance implications of the change 🙏🏻. 

The thread at https://github.com/iterative/studio/pull/8490#discussion_r1419809476 is relevant.

### CODE REVIEW

**Review by julieg18 — APPROVED** [review]
Looks good to me!

File: webview/src/plots/hooks/useGetPlot.ts [hunk 1]
```diff
@@ -9,49 +9,49 @@ import { fillTemplate } from '../components/vegaLite/util'
 export const useGetPlot = (
   section: PlotsSection,
   id: string,
-  plotRef: RefObject<HTMLButtonElement | HTMLDivElement>
+  plotRef: RefObject<HTMLButtonElement | HTMLDivElement>,
+  plotFocused: boolean
```
Comment (sroy3) [inline webview/src/plots/hooks/useGetPlot.ts:13]: Was this needed in the end?

**Comment by sroy3** [comment]
> @sroy3 when you're back I'd like to get your opinion on performance implications of the change 🙏🏻.

Since this applies to only one element and resizing is quite rare or happens at one moment only, the performance optimization here is really, really small, if there is one at all. But we should favour `ResizeObserver` and make it best practice as it is generally more performant and better suited for most uses.

File: webview/src/plots/hooks/useGetPlot.ts (hunk 1 above)
Comment (mattseddon) [reply webview/src/plots/hooks/useGetPlot.ts:13]: Yes, for suppressing the legend

**Pull request merged**